### PR TITLE
Improve IdP Integration Tests

### PR DESCRIPTION
**Pull request by Shenali-SJ**
### Proposed changes in this pull request

This PR improves integration tests related to Identity Providers.

**Related Issue:**
- https://github.com/wso2/product-is/issues/22001

---

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 1]
```diff
@@ -647,6 +707,7 @@ public void testGetIdPFederatedAuthenticators() throws Exception {
                 .assertThat()
                 .statusCode(HttpStatus.SC_OK)
                 .body(baseIdentifier + "name", equalTo("GoogleOIDCAuthenticator"))
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:709]: check authenticatorId as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 2]
```diff
@@ -552,8 +608,11 @@ public void testGetIdP() throws IOException {
                 .body("description", equalTo("IDP for Google Federation"))
                 .body("isEnabled", equalTo(true))
                 .body("isPrimary", equalTo(false))
-                .body("federatedAuthenticators.authenticators.find { it.authenticatorId == '" +
-                        SAMPLE_FEDERATED_AUTHENTICATOR_ID + "' }.definedBy", equalTo("SYSTEM"))
+                .body(baseIdentifier + "name", equalTo(new String(Base64.getDecoder().
+                        decode(SAMPLE_FEDERATED_AUTHENTICATOR_ID))))
+                .body(baseIdentifier + "isEnabled", equalTo(true))
+                .body(baseIdentifier + "definedBy", equalTo(DEFINED_BY_SYSTEM))
+                .body(baseIdentifier + "tags", Matchers.hasItems("Social-Login", "APIAuth"))
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:615]: check self as well

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 3]
```diff
@@ -529,7 +576,13 @@ public void addSAMLStandardBasedIdP() throws IOException {
...
                 .assertThat()
                 .statusCode(HttpStatus.SC_CREATED)
-                .header(HttpHeaders.LOCATION, notNullValue());
+                .header(HttpHeaders.LOCATION, notNullValue())
+                .body("name", equalTo(SAML_IDP_NAME))
+                .body(baseIdentifier + "authenticatorId", equalTo(SAML_AUTHENTICATOR_ID))
+                .body(baseIdentifier + "name", equalTo(new String(Base64.getDecoder().decode(SAML_AUTHENTICATOR_ID))))
+                .body(baseIdentifier + "isEnabled", equalTo(true))
+                .body(baseIdentifier + "definedBy", equalTo(DEFINED_BY_SYSTEM))
+                .body(baseIdentifier + "tags", Matchers.hasItems("SAML"));
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:585]: assert self as well

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 4]
```diff
@@ -431,7 +451,13 @@ public void testAddIdP() throws IOException {
...
                 .assertThat()
                 .statusCode(HttpStatus.SC_CREATED)
-                .header(HttpHeaders.LOCATION, notNullValue());
+                .header(HttpHeaders.LOCATION, notNullValue())
+                .body("name", equalTo(IDP_NAME))
+                .body(baseIdentifier + "name", equalTo(new String(Base64.getDecoder().
+                        decode(SAMPLE_FEDERATED_AUTHENTICATOR_ID))))
+                .body(baseIdentifier + "isEnabled", equalTo(true))
+                .body(baseIdentifier + "definedBy", equalTo(DEFINED_BY_SYSTEM))
+                .body(baseIdentifier + "tags", Matchers.hasItems("Social-Login", "APIAuth"));
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:461]: Assert self.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 5]
```diff
@@ -339,17 +343,27 @@ public void testGetMetaOutboundConnector() throws IOException {
...
                 .assertThat()
                 .statusCode(HttpStatus.SC_CREATED)
-                .header(HttpHeaders.LOCATION, notNullValue());
+                .header(HttpHeaders.LOCATION, notNullValue())
+                .body("name", equalTo(CUSTOM_IDP_NAME))
+                .body(baseIdentifier + "authenticatorId", equalTo(FEDERATED_AUTHENTICATOR_ID))
+                .body(baseIdentifier + "name", equalTo(new String(Base64.getDecoder().decode(FEDERATED_AUTHENTICATOR_ID))))
+                .body(baseIdentifier + "isEnabled", equalTo(true))
+                .body(baseIdentifier + "definedBy", equalTo(DEFINED_BY_USER))
+                .body(baseIdentifier + "tags", Matchers.hasItems("Custom"));
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:367]: Assert self, check other places as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 1 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:713]: This is updated with 7831a169607232cefc26307b20c8ca3746542a52

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 2 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:616]: This is updated with 7831a169607232cefc26307b20c8ca3746542a52

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 3 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java]: This is updated with 7831a169607232cefc26307b20c8ca3746542a52

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 6]
```diff
@@ -392,7 +410,13 @@ public void testUpdateUserDefinedAuthenticatorOfIdP() throws JsonProcessingExcep
                 .statusCode(HttpStatus.SC_OK)
                 .body("authenticatorId", equalTo(FEDERATED_AUTHENTICATOR_ID))
                 .body("name", equalTo(new String(Base64.getDecoder().decode(FEDERATED_AUTHENTICATOR_ID))))
-                .body("definedBy", equalTo("USER"))
+                .body("isEnabled", equalTo(true))
+                .body("definedBy", equalTo(DEFINED_BY_USER))
+                .body( "tags", hasItems("Custom"))
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:415]: remove unnecessary space before "tags"

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java [hunk 7]
```diff
@@ -409,7 +412,11 @@ public void testUpdateUserDefinedAuthenticatorOfIdP() throws JsonProcessingExcep
                 .body("name", equalTo(new String(Base64.getDecoder().decode(FEDERATED_AUTHENTICATOR_ID))))
                 .body("isEnabled", equalTo(true))
                 .body("definedBy", equalTo(DEFINED_BY_USER))
-                .body( "tags", Matchers.hasItems("Custom"))
+                .body( "tags", hasItems("Custom"))
+                .body("self", equalTo(getTenantedRelativePath(
```
Comment (Thisara-Welmilla) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:416]: Update response does not have self, ot has only endpoint config.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 4 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:469]: This is updated with e73225fc2e8068e574a4280d5206f44f5aaf2ceb

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 5 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:370]: This is updated with e73225fc2e8068e574a4280d5206f44f5aaf2ceb

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 6 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java]: This is updated with 19889d29a846f9073e796922aee6f22da20d7303

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java (hunk 7 above)
Comment (Shenali-SJ) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java:412]: This is updated with 19889d29a846f9073e796922aee6f22da20d7303

**Pull request merged**